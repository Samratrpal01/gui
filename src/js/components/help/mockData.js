export const helpProps = {
  docsVersion: undefined,
  isHosted: false,
  menderVersion: 'master',
  isEnterprise: true,
  tenantCapabilities: {},
  tokens: [],
  versions: {
    repos: {
      mender: '3.3.0',
      'mender-artifact': '3.8.0',
      'mender-binary-delta': '1.3.1',
      'mender-cli': '1.8.0',
      'mender-configure-module': '1.0.4',
      'mender-connect': '2.0.2',
      'mender-convert': '3.0.0',
      'mender-gateway': '1.0.0',
      'monitor-client': '1.2.0'
    },
    releaseDate: '01-01-2003'
  }
};
