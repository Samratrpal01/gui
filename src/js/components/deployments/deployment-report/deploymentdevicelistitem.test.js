import React from 'react';

import { adminUserCapabilities, defaultState, undefineds } from '../../../../../tests/mockData';
import { render } from '../../../../../tests/setupTests';
import DeploymentDeviceListItem from './deploymentdevicelistitem';

describe('DeploymentDeviceListItem Component', () => {
  it('renders correctly', async () => {
    const { baseElement } = render(
      <DeploymentDeviceListItem
        device={defaultState.deployments.byId.d1.devices.a1}
        idAttribute="Device ID"
        viewLog={jest.fn}
        userCapabilities={adminUserCapabilities}
      />
    );
    const view = baseElement.firstChild.firstChild;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
});
