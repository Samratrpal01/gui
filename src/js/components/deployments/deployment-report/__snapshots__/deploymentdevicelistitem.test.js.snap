// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeploymentDeviceListItem Component renders correctly 1`] = `
@keyframes animation-0 {
  0% {
    left: -35%;
    right: 100%;
  }

  60% {
    left: 100%;
    right: -90%;
  }

  100% {
    left: 100%;
    right: -90%;
  }
}

@keyframes animation-1 {
  0% {
    left: -200%;
    right: 100%;
  }

  60% {
    left: 107%;
    right: -8%;
  }

  100% {
    left: 107%;
    right: -8%;
  }
}

.emotion-0 {
  color: inherit;
  display: table-row;
  vertical-align: middle;
  outline: 0;
}

.emotion-0.MuiTableRow-hover:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

.emotion-0.Mui-selected {
  background-color: rgba(51, 122, 135, 0.08);
}

.emotion-0.Mui-selected:hover {
  background-color: rgba(51, 122, 135, 0.12);
}

.emotion-1 {
  font-family: Lato,sans-serif;
  font-weight: 400;
  font-size: 0.8125rem;
  line-height: 1.43;
  display: table-cell;
  vertical-align: inherit;
  border-bottom: 1px solid rgba(224, 224, 224, 1);
  text-align: left;
  padding: 16px;
  padding: 0px 24px 0px 24px;
  height: 48px;
}

.emotion-7 {
  position: relative;
  overflow: hidden;
  display: block;
  height: 4px;
  z-index: 0;
  background-color: rgb(177, 204, 209);
}

@media print {
  .emotion-7 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-8 {
  width: 100%;
  position: absolute;
  left: 0;
  bottom: 0;
  top: 0;
  -webkit-transition: -webkit-transform 0.2s linear;
  transition: transform 0.2s linear;
  transform-origin: left;
  background-color: #337a87;
  width: auto;
  -webkit-animation: animation-0 2.1s cubic-bezier(0.65, 0.815, 0.735, 0.395) infinite;
  animation: animation-0 2.1s cubic-bezier(0.65, 0.815, 0.735, 0.395) infinite;
}

.emotion-9 {
  width: 100%;
  position: absolute;
  left: 0;
  bottom: 0;
  top: 0;
  -webkit-transition: -webkit-transform 0.2s linear;
  transition: transform 0.2s linear;
  transform-origin: left;
  background-color: #337a87;
  width: auto;
  -webkit-animation: animation-1 2.1s cubic-bezier(0.165, 0.84, 0.44, 1) 1.15s infinite;
  animation: animation-1 2.1s cubic-bezier(0.165, 0.84, 0.44, 1) 1.15s infinite;
}

<tr
  class="MuiTableRow-root emotion-0"
>
  <td
    class="MuiTableCell-root MuiTableCell-sizeMedium emotion-1"
  >
    <a
      href="/devices?id=a1"
      style="font-weight: initial; opacity: 1;"
    >
      <div
        class="flexbox space-between"
      >
        <div>
          a1
        </div>
      </div>
    </a>
  </td>
  <td
    class="MuiTableCell-root MuiTableCell-sizeMedium emotion-1"
  >
    -
  </td>
  <td
    class="MuiTableCell-root MuiTableCell-sizeMedium emotion-1"
  >
    -
  </td>
  <td
    class="MuiTableCell-root MuiTableCell-sizeMedium emotion-1"
  >
    <div
      class=""
    >
      -
    </div>
  </td>
  <td
    class="MuiTableCell-root MuiTableCell-sizeMedium emotion-1"
  >
    <div
      class=""
    >
      -
    </div>
  </td>
  <td
    class="MuiTableCell-root MuiTableCell-sizeMedium emotion-1"
    style="padding-right: 0px; position: relative; min-width: 200px;"
  >
    installing
    <div
      style="position: absolute; bottom: 0px; width: 100%;"
    >
      <span
        class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate emotion-7"
        role="progressbar"
      >
        <span
          class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate emotion-8"
        />
        <span
          class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar2Indeterminate emotion-9"
        />
      </span>
    </div>
  </td>
  <td
    class="MuiTableCell-root MuiTableCell-sizeMedium emotion-1"
  />
</tr>
`;
